crates/nu-cli/src/cli.rs

```diff
@@ -789,7 +789,18 @@ pub async fn cli(
 
         let cwd = context.shell_manager.path();
 
-        rl.set_helper(Some(crate::shell::Helper::new(context.clone())));
+        let disable_hinter = config::config(Tag::unknown())?
+            .get("disable_hinter")
```

Comment:
Nice!

I think the one thing I might change is that rather than having it as "disable_hinter" at the very top-level of the config, move it into the "line_editor" section.

For example: https://github.com/nushell/nushell/blob/48850becd8270bb653a6681de8e4eec00f4c157a/crates/nu-cli/src/cli.rs#L605

You wouldn't need to move the logic. You could just look in a different part of the config.

Reply:
+1 for having it in the line editor section.

I also would love for configuration options to be "positive" forms. In this case, I would call the configuration option `show_hints`, and default it to `true`. Negative forms are slightly more work to think about and are slightly less useful (see my other comment).

Reply:
Done. Currently all the "line_editor" config is for rustyline and reusing the code that sets those config is a bit inconvenient. But conceptually it makes sense to be inside that section. Also flipped the negative and made it `show_hints`.

Reply:
Ah yes, that does kind of suck, but agreed that we're aiming for it to make sense conceptually. One day we may not have rustyline, and arguably all of these configs should still apply to that new thing.